frontends/parsers/p4/p4parser.ypp

```diff
@@ -1074,10 +1074,15 @@ enumDeclaration
     : optAnnotations
         ENUM name { driver.structure->declareType(*$3); }
         "{" identifierList "}" { $$ = new IR::Type_Enum(@3, *$3, *$6); }
-    | optAnnotations ENUM BIT l_angle INTEGER r_angle name { driver.structure->declareType(*$7); }
-        "{" specifiedIdentifierList "}" {
-            auto type = IR::Type::Bits::get(@3+@6, parseConstantChecked(@5, $5));
-            $$ = new IR::Type_SerEnum(@7, *$7, type, *$10); }
+    | optAnnotations ENUM typeRef name { driver.structure->declareType(*$4); }
+	 "{" specifiedIdentifierList "}" {
+              auto type = $typeRef;
+              if (auto tn = type->to<IR::Type_Name>()) {
```

Comment:
You should not check the type here; this should be left for the typechecker.
Also, this is wrong, because if neither condition is true $$ will be left uninitialized.